hostprocess/Install-Containerd.ps1

```diff
@@ -39,30 +45,56 @@ function DownloadFile($destination, $source) {
     }
 }
 
-$requiredWindowsFeatures = @(
-    "Containers",
-    "Hyper-V",
-    "Hyper-V-PowerShell")
-
-function ValidateWindowsFeatures {
-    $allFeaturesInstalled = $true
-    foreach ($feature in $requiredWindowsFeatures) {
-        $f = Get-WindowsFeature -Name $feature
-        if (-not $f.Installed) {
-            Write-Warning "Windows feature: '$feature' is not installed."
-            $allFeaturesInstalled = $false
-        }
-    }
-    return $allFeaturesInstalled
+$rebootNeeded = $false
+
+$windowsFeatures = @("Containers")
+
+if (!$skipHypervisorSupportCheck) {
+    $windowsFeatures += @("Hyper-V", "Hyper-V-PowerShell")
 }
 
-if (-not (ValidateWindowsFeatures)) {
-    Write-Output "Installing required windows features..."
+foreach ($feature in $windowsFeatures) {
+    $featureInstalled = (Get-WindowsFeature -Name $feature).Installed
 
-    foreach ($feature in $requiredWindowsFeatures) {
+    if ($featureInstalled) {
+        Write-Output "Windows feature '$feature' is already installed."
+    }
+    else {
+        Write-Warning "Windows feature '$feature' is not installed. Installing it now..."
         Install-WindowsFeature -Name $feature
+        $rebootNeeded = $true
+    }
+}
+
+if ($skipHypervisorSupportCheck) {
+    $hyperVFeatureName = "Microsoft-Hyper-V"
+    $hyperVState = (Get-WindowsOptionalFeature -FeatureName $hyperVFeatureName -Online).State
+
+    if ($hyperVState -eq "Enabled") {
+        Write-Output "Windows optional feature '$hyperVFeatureName' is already enabled."
+    }
+    else {
+        Write-Warning "Windows optional feature '$hyperVFeatureName' is not enabled. Enabling it now..."
+        DISM /Online /Enable-Feature /FeatureName:$hyperVFeatureName /All /NoRestart
+        $rebootNeeded = $true
+    }
+
+    $hyperVOnlineFeatureName = "Microsoft-Hyper-V-Online"
+    $hyperVOnlineState = (Get-WindowsOptionalFeature -FeatureName $hyperVOnlineFeatureName -Online).State
+
+    if ($hyperVOnlineState -eq "Disabled") {
+        Write-Output "Windows optional feature '$hyperVOnlineFeatureName' is already disabled."
+    }
+    else {
+        Write-Warning "Windows optional feature '$hyperVOnlineFeatureName' is not disabled. Disabling it now..."
+        DISM /Online /Disable-Feature /FeatureName:$hyperVOnlineFeatureName /NoRestart
+        $rebootNeeded = $true
     }
 
+    Write-Warning "The Hyper-V features was installed without checking the CPU for Hypervisor support. You way wont be able to host Hyper-V isolated containers."
```

Comment:
```suggestion
    Write-Warning "The Hyper-V features was installed without checking the CPU for Hypervisor support. You may not be able to host Hyper-V isolated containers if CPU doesn't support hypervisors."
```

Reply:
is this correct? It looks like in all cases we disabled the hypervisor?  I think we should just say hypervisor isn't not enabled.

Reply:
I think so, yes. This is what I understood from the link you provided: https://github.com/kubernetes-sigs/image-builder/blob/3c9c5642ac78be02ff631a19fef6fc3dab4c1a90/images/capi/ansible/windows/roles/systemprep/tasks/main.yml#L157-L176

Also, we disable it only if `skipHypervisorSupportCheck` is set and if it's not disabled already.